src/ert/config/observations.py

```diff
@@ -61,8 +80,12 @@ def __getitem__(self, key: str) -> ObsVector:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, EnkfObs):
             return False
+
+        if self.datasets.keys() != other.datasets.keys():
```

Comment:
Isnt this duplicated in `ErtConfig`?

Reply:
Appears so, but this is for the `EnkfObs`, and in `ErtConfig` it is for the dict mapping response type to obs ds. Long-term we should maybe cut out enkfobs and only keep the dict but right now it is a bit duplicated and necessary.